analysis/quantifier-comparisons/12-spatial-transcriptomics-storage.Rmd

```diff
@@ -0,0 +1,206 @@
+image_data <- image_data[common_cells,]
+
+# add image data to Seurat
+Seurat::DefaultAssay(object = image_data) <- "Spatial"
+fry_seurat[['slice']] <- image_data
+```
+
+
+### Comparing Seurat and Spatial Experiment
+
+Now let's take a look at how each of these objects stores information for the experiment. 
+
+```{r}
+head(colData(fry_spe))
+head(fry_seurat@meta.data)
+```
+
+Seurat automatically adds some per cell QC metrics here, but in general the `colData` looks similar to what we would expect for a `SingleCellExperiment` for the `SpatialExperiment`.
+
+```{r}
+head(rowData(fry_spe))
+head(fry_seurat[["Spatial"]]@meta.features)
+```
+
+It looks like both of these start off with no `rowData` as expected.
+
+```{r}
+fry_spe <- fry_spe %>%
+  scuttle::addPerCellQCMetrics(subsets = list(mito_genes = mito_genes[mito_genes %in% rownames(fry_spe)])) 
+head(colData(fry_spe))
+```
+
+You can still use the same functions to add in per cell metrics that we are using in `scpcaTools` now on a `SpatialExperiment`. 
+
+Let's specifically look at how each of them stores the data related to the spatial information.
+The `SpatialExperiment` has three places where it stores spatial information, `spatialData`, `spatialCoords`, and `imgData`. 
+
+```{r}
+head(spatialData(fry_spe))
+```
+
+```{r}
+head(spatialCoords(fry_spe))
+```
+
+`spatialData` holds information about whether or not each barcode was covered by tissue and spatial metadata.
+`spatialCoords` holds the coordinates for each spot.
+
+This same information can be found by looking in the `images` slot in the `Seurat` object. 
+
+```{r}
+head(fry_seurat@images)
+```
+
+`imgData` holds information about each image that is stored as multiple slices of an image can be stored within one `SpatialExperiment`. 
+
+```{r}
+imgData(fry_spe)
+```
+
+This same information can be found by looking at the `images` slot in the `Seurat` object. 
+
+```{r}
+fry_seurat@images$slice@scale.factors
+```
+
+Let's take a look at some example plots that can be produced with each of these objects.
+
+```{r}
+library(viridis)
+# plot tissue spots coloring by total UMI count
+ggspavis::plotSpots(spe = fry_spe, 
+          x_coord = "pxl_col_in_fullres", 
+          y_coord = "pxl_row_in_fullres", 
+          annotate = "sum") +
+  scale_color_viridis_c()
+```
+You can also overlay the image by using `plotVisium`. 
+
+```{r}
+ggspavis::plotVisium(spe = fry_spe[,spatialData(fry_spe)$in_tissue == 1],
+                     x_coord = "pxl_col_in_fullres",
+                     y_coord = "pxl_row_in_fullres")
+```
+
+```{r}
+Seurat::SpatialFeaturePlot(fry_seurat, features = "nCount_Spatial")
+```
+
+It looks like overall they both would work and that the `SpatialExperiment` will provide similar functionality to how we have things already set up for single-cell experiments. 
+One thing to note is that you cannot have `SpatialExperiment` installed at the same time that you attempt to do this plot. 
```

Comment:
I feel like I need more explanation here. Does this notebook not run if executed from top to bottom? I see the error about `SpatialImage` being found from Seurat, but is that affecting the running of this notebook directly?

If so, I think we need to figure out how to communicate that and/or include the installation/uninstallation in the notebook directly. 

Fundamentally, I think it is quite important to say directly that having both Seurat and SpatialExperiment installed (even if not loaded!) is a problem. In my mind, this is almost a dealbreaker.